optimum/exporters/executorch/tasks/multimodal_text_to_text.py

```diff
@@ -0,0 +1,179 @@
+                - cache_implementation (str, optional):
+                    Cache management strategy (default: "static").
+                - max_length (int, optional):
+                    Maximum sequence length for generation (default: 2048).
+
+    Returns:
+        MultiModalTextToTextExportableModule:
+            An instance of `MultiModalTextToTextExportableModule` for exporting and lowering to ExecuTorch.
+    """
+    device = "cpu"
+    batch_size = 1
+    dtype = kwargs.get("dtype", "float32")
+    use_custom_sdpa = kwargs.get("use_custom_sdpa", False)
+    use_custom_kv_cache = kwargs.get("use_custom_kv_cache", False)
+    attn_implementation = kwargs.get("attn_implementation", "custom_sdpa" if use_custom_sdpa else "sdpa")
+    cache_implementation = kwargs.get("cache_implementation", "static")
+    use_custom_sdpa = use_custom_sdpa or attn_implementation == "custom_sdpa"
+    max_length = kwargs.get("max_length", 2048)
+    config = kwargs.get("config") or AutoConfig.from_pretrained(model_name_or_path)
+
+    # Make sure config has text_config and vision_config:
+    if not (hasattr(config, "text_config") and (hasattr(config, "vision_config") or hasattr(config, "audio_config"))):
+        raise ValueError(
+            f"The model {model_name_or_path} does not have a `text_config` or `vision_config`/`audio_config` attribute in its config. "
+            "This is required for multimodal text-to-text models."
+        )
+
+    if hasattr(config, "rope_scaling") and config.rope_scaling is not None:
+        # NOTE: Avoid hitting the data-dependent control flow in _longrope_frequency_update.
+        config.rope_scaling["type"] = "default"
+    if hasattr(config, "use_cache") and config.use_cache is False:
+        config.use_cache = True
+
+    eager_model = AutoModel.from_pretrained(
+        model_name_or_path,
+        device_map=device,
+        torch_dtype=dtype,
+        config=config,
+        attn_implementation=attn_implementation,
+        generation_config=GenerationConfig(
+            use_cache=True,
+            cache_implementation=cache_implementation,
+            max_length=max_length,
+            cache_config={
+                "batch_size": batch_size,
+                "max_cache_len": max_length,
+            },
+        ),
+    )
+    decoder_name, audio_encoder_name, vision_encoder_name = _validate_multimodal_components(eager_model)
+    # Need to do this since apparently when nested modules (e.g. model.language_model) access the .property
+    # config, it always comes from the generation_config.json file, not the `generation_config` override
+    # from from_pretrained().
+    getattr(eager_model, decoder_name).generation_config = eager_model.generation_config
+
+    for param in eager_model.parameters():
+        # Must disable gradient for quantized checkpoint
+        if isinstance(param, torchao.utils.TorchAOBaseTensor):
+            param.requires_grad = False
```

Comment:
nit: probably can be refactored to some utils. i see other tasks doing similar stuff